### PR TITLE
OBGM-565 Fix accessing nested map values

### PR DESCRIPTION
**Pull request by drodzewicz**

### CODE REVIEW

**Review by alannadolny — APPROVED** [review]

**Review by awalkowiak — APPROVED** [review]

File: grails-app/controllers/org/pih/warehouse/api/ReplenishmentApiController.groovy [hunk 1]
```diff
@@ -57,7 +57,7 @@ class ReplenishmentApiController {
 
         bindReplenishmentData(replenishment, currentUser, currentLocation, jsonObject)
         Order order = replenishmentService.createOrUpdateOrderFromReplenishment(replenishment)
-        if (order.hasErrors() || !order.save(flush: true)) {
+        if (order.hasErrors()) {
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/api/ReplenishmentApiController.groovy:60]: We should not be doing that check here in the controller. I think I saw that this was moved to the service in another PR and I just wanted to mention that I think is the proper approach.

File: grails-app/controllers/org/pih/warehouse/api/ReplenishmentApiController.groovy [hunk 2]
```diff
@@ -99,7 +99,7 @@ class ReplenishmentApiController {
     }
 
     Replenishment bindReplenishmentData(Replenishment replenishment, User currentUser, Location currentLocation, JSONObject jsonObject) {
-        bindData(replenishment, jsonObject)
+        bindData(replenishment, jsonObject, [exclude: ['replenishmentItems']])
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/api/ReplenishmentApiController.groovy:102]: A comment would be helpful here so we know why the exclude is being used.

**Review by jmiranda — COMMENTED** [review]
Nice work, @drodzewicz
